pkg/compression/compression.go

```diff
@@ -165,3 +171,57 @@ func AutoDecompress(stream io.Reader) (io.ReadCloser, bool, error) {
 	}
 	return res, decompressor != nil, nil
 }
+
+type estarCompressor struct {
```

Comment:
Move this into a separate file, please; let’s keep `compression.go` somewhat generic.